src/components/Writing-Assistant/WritingAssistantFloatingMenu.tsx

```diff
@@ -0,0 +1,446 @@
+        currentChatHistory.displayableChatHistory.length - 1
+      ];
+
+    const insertionText = llmResponse.visibleContent
+      ? llmResponse.visibleContent
+      : formatOpenAIMessageContentIntoString(llmResponse.content);
+
+    editor.view.focus();
+
+    const { from, to } = editor.state.selection;
+    const endOfSelection = Math.max(from, to);
+
+    editor
+      .chain()
+      .focus()
+      .setTextSelection(endOfSelection)
+      .insertContent("\n" + insertionText)
+      .run();
+
+    setCurrentChatHistory(undefined);
+  };
+
+  const replaceHighlightedText = () => {
+    if (
+      !editor ||
+      !currentChatHistory ||
+      currentChatHistory.displayableChatHistory.length === 0
+    ) {
+      console.error("No chat history available for replacement.");
+      return;
+    }
+
+    const llmResponse =
+      currentChatHistory.displayableChatHistory[
+        currentChatHistory.displayableChatHistory.length - 1
+      ];
+
+    const replacementText = llmResponse.visibleContent
+      ? llmResponse.visibleContent
+      : formatOpenAIMessageContentIntoString(llmResponse.content);
+
+    editor
+      .chain()
+      .focus()
+      .deleteSelection()
+      .insertContent(replacementText)
+      .run();
+
+    setCurrentChatHistory(undefined);
+  };
+
+  const handleOption = async (option: string, customPromptInput?: string) => {
+    const selectedText = highlightData.text;
+    if (!selectedText.trim()) return;
+
+    let prompt = "";
+
+    switch (option) {
+      case "simplify":
+        prompt = `The following text in triple quotes below has already been written:
+"""
+${selectedText}
+"""
+Simplify and condense the writing. Do not return anything other than the simplified writing. Do not wrap responses in quotes.`;
+        break;
+      case "copy-editor":
+        prompt = `Act as a copy editor. Go through the text in triple quotes below. Edit it for spelling mistakes, grammar issues, punctuation, and generally for readability and flow. Format the text into appropriately sized paragraphs. Make your best effort.
+ 
+""" ${selectedText} """
+Return only the edited text. Do not wrap your response in quotes. Do not offer anything else other than the edited text in the response. Do not translate the text. If in doubt, or you can't make edits, just return the original text.`;
+        break;
+      case "takeaways":
+        prompt = `My notes are below in triple quotes:
+""" ${selectedText} """
+Write a markdown list (using dashes) of key takeaways from my notes. Write at least 3 items, but write more if the text requires it. Be very detailed and don't leave any information out. Do not wrap responses in quotes.`;
+        break;
+      case "custom":
+        prompt =
+          `prompt(in triple #): ` +
+            `### ` +
+            customPromptInput +
+            ` ###` +
+            ` ,apply the prompt to the text in triple quotes """ ${selectedText} """` ||
```

Comment:
This prompt imo could be improved to be like "The user has given the following instructions for processing the text selected" or something like htat

Reply:
modified